Expand CRAM index section.

jkbonfield opened the pull request:
Fixes #376

jmthibault79 commented on CRAMv3.tex:
```diff
@@ -1539,31 +1542,80 @@ \subsubsection*{General notes}
+Each line represents a slice in the CRAM file.
+Please note that all slices must be listed in the index file.
+
+Multi-reference slices must may need to have multiple lines for the same slice; one for each reference contained within that slice.
```
rm "must"

cmnbroad commented on CRAMv3.tex:
```diff
@@ -1539,31 +1542,80 @@ \subsubsection*{General notes}
+\draw[|<->|] (9.5,1.8) -- (11.5,1.8);
+\node [above] at (6.5,1.8) {Size 1};
+\node [above] at (8.5,1.8) {Size 2};
+\node [above] at (10.5,1.8){Size 3};
```
Nice!

jkbonfield replied:
It's my first time exploring pgf/tikz.  I already see I can write this in a friendlier fashion, but it's a good start and I like the ability to version control our diagrams.

jmarshall replied:
Yes indeed :smile:

Trivially, perhaps wrap `\begin{center}` around it? That appears to be the way to centre a `tikzpicture`…

jkbonfield replied:
Now centred, and rewritten the pic layout.

Useful resources I found:

http://www-groups.mcs.st-andrews.ac.uk/~alanc/pub/c_tikzref/c_tikzref.pdf

https://stuff.mit.edu/afs/athena/contrib/tex-contrib/beamer/pgf-1.01/doc/generic/pgf/version-for-tex4ht/en/pgfmanualse8.html

jkbonfield commented on CRAMv3.tex:
```diff
@@ -1529,6 +1530,8 @@ \section{\textbf{Indexing}}
 
 \subsubsection*{General notes}
 
+Indexing is only valid on coordinate (reference and position) sorted files.
```
Agreed we should be consistent here with SAM wording.  Better to just copy the explanation instead of linking to an external reference.

jkbonfield commented on CRAMv3.tex:
```diff
@@ -1539,31 +1542,80 @@ \subsubsection*{General notes}
+In this case the index reference sequence ID will be the actual reference number (from the ``RI'' data series) and not -2.
+
+Slices containing solely unmapped unplaced data (reference ID -1) still require values for all columns, although the alignment start and span will be ignored.
+It is recommended that they are set to small constant values (0 or 1).
```
Rather embarrassingly I put 0 or 1 because my own code uses both. ;-)

It doesn't actually matter as we define them to be ignored on read, but a constant single byte value keeps the index small and compressible.  I'll just change it to zero I think and at some point fix my code.

jkbonfield commented:
Updated following review, thanks.

I added an extra commit to adjust the one remaining use of "Container structure" as it seems pointless to need a separate PR for this, although when squashing we should probably retain that as its own commit.
